.github/workflows/linux_and_macos.yml

```diff
@@ -99,6 +111,51 @@ jobs:
           make uninstall DESTDIR="${PWD}"/ROOT/
           [[ "$(find ROOT/ -not -type d | tee /dev/stderr)" == '' ]]  # i.e. fail CI if leftover files
 
+      - name: 'Run UI tests'
+        if: "${{ runner.os == 'macOS' }}"
+        run: |-
+          ./recordings/record.sh
+
+      - name: 'Upload UI test renderings for inspection'
+        if: "${{ runner.os == 'macOS' }}"
+        uses: actions/upload-artifact@a8a3f3ad30e3422c9c7b888a15615d19a852ae32  # v3.1.3
+        with:
+          name: ttyplot_ui_test_${{ matrix.runs-on }}_${{ github.sha }}
+          path: recordings/actual*
+          if-no-files-found: error
+
+      - name: 'Evaluate UI test results'
+        if: "${{ runner.os == 'macOS' }}"
+        run: |-
+          set -o pipefail
+
+          assert_images_equal_enough() {
+            local a="${1}"
+            local b="${2}"
+            local diff_output="${3}"
+            local dissimilarity="$(compare -metric DSSIM "${a}" "${b}" "${diff_output}" 2>&1)"
+            if ! python3 <<<"import sys; sys.exit(int(${dissimilarity} > 0.0022))"; then
```

Comment:
Maybe `bc` could be a lightweight alternative to Python:

```shell
if [ $(bc <<< "$dissimilarity > 0.0022") = "1" ]; then ...
```

Reply:
@edgar-bonet I started out with `bc` in fact, but it turned out that ImageMagick's `compare` sometimes produces scientific output like `7.57917e-05` and `bc` fails to parse that (see below) while Python 3 has no problem with it:

```console
# bc <<<"7.57917e-05 < 123"
(standard_in) 1: syntax error

# python3 <<<"print(7.57917e-05 < 123)"
True
```

That's the only reason I went for Python there.